pkg/controller/clusterconnection/clusterconnection_controller.go

```diff
@@ -461,3 +466,45 @@ func (r *ReconcileConnection) Reconcile(ctx context.Context, request reconcile.R
 	// We should create the Guardian deployment.
 	return result, nil
 }
+
+// maintainInstallationFinalizer manages this controller's finalizer on the Installation resource.
+// We add a finalizer to the Installation when Guardian has been installed, and only remove that finalizer when
+// the Guardian has been deleted and its pods have stopped running. This allows for a graceful cleanup of Guardian resources
+// prior to the CNI plugin being removed.
+func maintainInstallationFinalizer(ctx context.Context, c client.Client, mgmtClusterConnection *operatorv1.ManagementClusterConnection) error {
```

Comment:
Now that we're using this across many controllers, I think it is time we make this a common generic utility function rather than duplicate the code in each controller.